automation_tools/__init__.py

```diff
@@ -1465,3 +1469,178 @@ def setenforce(mode):
         ))
 
     run('setenforce {0}'.format(mode))
+
+
+def get_hostname_from_ip(ip):
+    """Retrives the hostname by logging into remote machine by IP.
+    Specially for the systems who doesnt support reverse DNS.
+    e.g usersys machines.
+
+    :param str ip: Remote host's IP address to retrive hostname.
+    """
+    gethost = lambda: run('hostname')
+    result = execute(gethost, host=ip)
+    print('host', result[ip])
+    return result[ip]
+
+
+def create_openstack_instance(
+        instance_name=None, image_name=None, flavor_name=None, ssh_key=None):
+    """Creates openstack Instance from Image and Assigns a floating IP \
+    to instance.
+
+    :param str instance_name: Openstack Instance name to create.
+    :param str image_name: Openstack image name from which \
+    instance to be created.
+    :param str flavor_name: Openstack flavor_name for instance. e.g m1.small.
+    :param str ssh_key: ssh_key 'name' that you want to add to this instance.
+
+    ssh_key should be added to openstack project before running automation.
+    Else the automation will fail.
+    """
+    if (instance_name or image_name or flavor_name or ssh_key) is None:
+        raise ValueError(
+            'instance_name/image_name/flavor_name/ssh_key cannot be None.')
+    network_name = 'satellite-jenkins'
+    clientObj = client.Client(
```

Comment:
Using openstack_client instead, as this is not specific for satellite product. 
It will be used for satellite, capsule as well as for client upgrade.